sample_apps/run_example.sh

```diff
@@ -392,20 +418,28 @@ function run_cmd() {
     echo " "
     This_fn="${FUNCNAME[1]}"
     if [ "${Prev_fn}" != "${This_fn}" ]; then
-        echo "******************************************************************************"
-        echo "${Me}: Running ${FUNCNAME[1]} "
+        echo "**************************************************************************************************************"
+        echo "${Me}: ${RunTag}: Running${DryRunTag}${FUNCNAME[1]} "
         Prev_fn="${This_fn}"
     fi
 
+    # For tracing, some functions simply invoke this run_cmd() with
+    # no arguments. Just 'sudo' will fail, so side-step this usage 'issue'.
+    local args_array=("$@")
+    local array_len=0
+    array_len=$((${#args_array[@]}))
+
+    local echo_sudo=""
+    local run_sudo=0
+    if [ "${RunSudo}" -eq 1 ] && [ "${array_len}" -gt "1" ]; then
+        echo_sudo="sudo "
+        run_sudo=1
+    fi
+
```

Comment:
You can safely ignore these diffs. These are solely for improved messages indicating what's being run in normal and in `--dry-run` mode.